File: stanford-algorithms/course-1-divide-and-conquer/week-2/bubblesort/bubblesort.py
def bubblesort(arr):
    result = arr.copy()
    size = len(arr)
    changed = False
    for i in range(size):
        for j in range(size-i):
            if result[j] > result[j+1]:
                result[j], result[j+1] = result[j+1], result[j]
                changed = True
        if not changed:
            break
    return result


def main():
    arr = [1, 46, 7865, 23, 12, 3]
    final = bubblesort(arr)
    print(f"Initial: {arr}")
    print(f"Final: {final}")

if __name__ == "__main__":
    main()

File: stanford-algorithms/course-1-divide-and-conquer/week-2/bubblesort/bubblesort.go
package main

import (
	"fmt"
)

func bubblesort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		changed := false
		for j := 0; j < len(arr)-i; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
				changed = true
			}
		}
		if !changed {
			break
		}
	}
	return arr
}

func main() {
	initial := []int{1, 2, 64, 23, 3, 8, 5, 7}
	fmt.Printf("Initial:\t\t%v\n", initial)
	bubblesort(initial)
	fmt.Printf("Final:\t\t\t%v", initial)
}
